front/components/vaults/AddConnectionMenu.tsx

```diff
@@ -0,0 +1,263 @@
+          } satisfies PostManagedDataSourceRequestBody),
+        }
+      );
+
+      if (res.ok) {
+        const createdManagedDataSource: {
+          dataSource: DataSourceType;
+          connector: ConnectorType;
+        } = await res.json();
+        //TODO(GROUPS-UI): redirect to new modal to edit permissions
+        void router.push(
+          `/w/${owner.sId}/builder/data-sources/${createdManagedDataSource.dataSource.name}` +
+            (REDIRECT_TO_EDIT_PERMISSIONS.includes(provider)
+              ? `?edit_permissions=true`
+              : "")
+        );
+      } else {
+        const responseText = await res.text();
+        sendNotification({
+          type: "error",
+          title: `Failed to enable connection (${provider})`,
+          description: `Got: ${responseText}`,
+        });
+      }
+    } catch (e) {
+      setShowConfirmConnection((prev) => ({
+        isOpen: false,
+        connector: prev.connector,
+      }));
+      sendNotification({
+        type: "error",
+        title: `Failed to enable connection (${provider})`,
+      });
+    } finally {
+      setIsProviderLoading(provider, false);
+    }
+  };
+
+  const handleConnectionClick = (connectorProvider: ConnectorProvider) => {
+    const configuration = CONNECTOR_CONFIGURATIONS[connectorProvider];
+
+    const isBuilt =
+      configuration.status === "built" ||
+      (configuration.status === "rolling_out" &&
+        !!configuration.rollingOutFlag &&
+        owner.flags.includes(configuration.rollingOutFlag));
+    const isProviderAllowed = isConnectorProviderAllowed(
+      configuration.connectorProvider,
+      plan.limits.connections
+    );
+
+    const existingDataSource = existingDataSources.find(
+      (view) => view.connectorProvider === connectorProvider
+    );
+    if (!existingDataSource || !existingDataSource.connector) {
+      if (!isProviderAllowed) {
+        setShowUpgradePopup(true);
+      } else {
+        if (isBuilt) {
+          setShowConfirmConnection({
+            isOpen: true,
+            connector: connectorProvider,
+          });
+        } else {
+          setShowPreviewPopupForProvider({
+            isOpen: true,
+            connector: connectorProvider,
+          });
+        }
+      }
+    }
+  };
+
+  return (
+    nonSetUpConnectors.length > 0 && (
+      <>
+        <Dialog
+          isOpen={showUpgradePopup}
+          onCancel={() => setShowUpgradePopup(false)}
+          title={`${plan.name} plan`}
+          onValidate={() => {
+            void router.push(`/w/${owner.sId}/subscription`);
+          }}
+        >
+          <p>Unlock this managed data source by upgrading your plan.</p>
+        </Dialog>
+
+        <CreateConnectionConfirmationModal
+          connectorProviderConfiguration={
+            showConfirmConnection.connector
+              ? CONNECTOR_CONFIGURATIONS[showConfirmConnection.connector]
+              : CONNECTOR_CONFIGURATIONS.confluence
```

Comment:
Why do we have a default on confluence here?

Reply:
modal is expecting non-null value - changed to not render modal id it's not set